### PR TITLE
Modify GeoJSON module

### PR DESCRIPTION
**Pull request by whuaegeanse**
Modify codes of GeoJSON module.
- Make use of right value reference for temp variables;
- Reserve memory of std::vector.

### CODE REVIEW

**Comment by pramsey** [comment]
Can you do this w/o all the formatting changes? Your actual changes are hard to locate in the PR.

**Comment by whuaegeanse** [comment]
All of changes in this PR have been formatted with AStyle of GEOS.
Some files in this PR does not change after using AStyle of GEOS.

**Comment by dbaston** [comment]
@pramsey see https://github.com/libgeos/geos/pull/466/files?w=1

**Comment by whuaegeanse** [comment]
> Can you do this w/o all the formatting changes? Your actual changes are hard to locate in the PR.

After searching on google, I learned how to use AStyle to format files on the windows system. Maybe we explain how to use AStyle formatting files under windows system, just like providing instructions for use under linux system.

**Comment by pramsey** [comment]
Well, this situation is complicated by the fact that the original author did not astyle their commit, so you astyling on top of his code doesn't result in a nice clean "changes only" result, it's still very noisy. I will just go ahead and try and review through the static.

File: src/io/GeoJSONReader.cpp [hunk 1]
```diff
@@ -45,210 +45,268 @@ GeoJSONReader::GeoJSONReader(): GeoJSONReader(*(GeometryFactory::getDefaultInsta
...
 }
 
-std::unique_ptr<geom::Polygon> GeoJSONReader::readPolygon(const std::vector<std::vector<std::vector<double>>>& polygonCoords) {
+std::unique_ptr<geom::Polygon> GeoJSONReader::readPolygon(
+    const std::vector<std::vector<std::vector<double>>>& polygonCoords) const
+{
     std::unique_ptr<geom::LinearRing> shell;
     std::vector<std::unique_ptr<geom::LinearRing>> rings;
-    for(const auto& ring : polygonCoords) {
+    rings.reserve(polygonCoords.size());
```
Comment (pramsey) [inline src/io/GeoJSONReader.cpp:259]: Do we have tests for the various cases where the ordinate arrays can be zero-length? Is reserving 0 acceptable behaviour?

File: src/io/GeoJSONWriter.cpp [hunk 2]
```diff
@@ -39,111 +39,132 @@ using json = geos_nlohmann::ordered_json;
...
     return geojson;
 }
 
-std::string GeoJSONWriter::writeFormatted(const geom::Geometry* geometry, GeoJSONType type, int indent) {
+std::string GeoJSONWriter::writeFormatted(const geom::Geometry* geometry, GeoJSONType type, int indent)
+{
     json j;
     encode(geometry, type, j);
-    std::string geojson = j.dump(indent);
+    const std::string& geojson = j.dump(indent);
```
Comment (pramsey) [inline src/io/GeoJSONWriter.cpp:54]: Just for my curiousity (I'm no cxx expert) is this really saving a copy (assigning to a const reference before returning) or is it just making us feel good about apparently saving a copy?

File: src/io/GeoJSONReader.cpp (hunk 1 above)
Comment (whuaegeanse) [reply src/io/GeoJSONReader.cpp:259]: > Do we have tests for the various cases where the ordinate arrays can be zero-length? Is reserving 0 acceptable behaviour?

Only the new capacity is greater than the existing capacity will trigger memory allocation.

File: src/io/GeoJSONWriter.cpp (hunk 2 above)
Comment (whuaegeanse) [reply src/io/GeoJSONWriter.cpp:54]: I have made a new  PR #467  to fix bug and enhance functions.

**Pull request merged**